### ISSUE TITLE
[🐛BUG] DIN AUC性能测试与原文不符

### ISSUE THREAD

**Issue by NEUYuYang** [RUCAIBox/RecBole#1914]
我在recbole框架中运行DIN模型时，使用ml-20m数据集，测试集的AUC可以达到98%以上，但是在DIN原文中只有0.7337，请问你们在与原文进行比较的时候参数是怎么设定的?   其他学者用ml-20m数据集跑DIN的时候可以达到跟原文一样的性能指标吗

**Comment by zhengbw0324** [comment]
@NEUYuYang 
您好！可以提供一些更详细的信息吗，比如数据集和模型的配置参数等。

**Comment by NEUYuYang** [comment]
数据集用了recbloe框架提供的ml-20m数据集，用了userid,itemid和timestamp
其他参数就是默认的
正常来说参数不好只能AUC效果低，但是我运行的时候AUC效果太高了。。。。&nbsp; &nbsp;98点几
原文AUC只有七十多

---原始邮件---
发件人: "Bowen ***@***.***&gt;
发送时间: 2023年11月16日(周四) 下午5:17
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [RUCAIBox/RecBole] [🐛BUG] DIN AUC性能测试与原文不符 (Issue #1914)




 
@NEUYuYang
 您好！可以提供一些更详细的信息吗，比如数据集和模型的配置参数等。

**Comment by zhengbw0324** [comment]
@NEUYuYang 
您好！跑CTR任务需要指定数据的label。不进行特殊设置的话数据的所有交互都会被默认为正例。您可以通过设置`threshold`参数格局rating值区分正负例，具体可以参考我们的[文档](https://recbole.io/docs/user_guide/config/data_settings.html)。一个例子是：
```yaml
field_separator: "\t"
seq_separator: " "
USER_ID_FIELD: user_id
ITEM_ID_FIELD: item_id
RATING_FIELD: rating
NEG_PREFIX: neg_
LABEL_FIELD: label
val_interval:
  rating: "[0,3);(3,inf)"
threshold:
    rating: 4
load_col:
    inter: [user_id, item_id, rating]
    user: [user_id, age, gender, occupation]
    item: [item_id, genre, release_year]
# training and evaluation
epochs: 500
train_batch_size: 4096
eval_batch_size: 40960000

eval_args:
  split: {'RS':[0.8, 0.1, 0.1]}
  group_by: ~
  mode: labeled
  order: RO
valid_metric: AUC
metrics: ['AUC', 'LogLoss']
```
您可以根据您的需求自行调整配置参数。

**Comment by NEUYuYang** [comment]
好滴，谢谢



---原始邮件---
发件人: "Bowen ***@***.***&gt;
发送时间: 2023年11月16日(周四) 下午5:44
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [RUCAIBox/RecBole] [🐛BUG] DIN AUC性能测试与原文不符 (Issue #1914)




 
@NEUYuYang
 您好！跑CTR任务需要指定数据的label。不进行特殊设置的话数据的所有交互都会被默认为正例。您可以通过设置threshold参数格局rating值区分正负例，具体可以参考我们的文档。一个例子是：
 field_separator: "\t" seq_separator: " " USER_ID_FIELD: user_id ITEM_ID_FIELD: item_id RATING_FIELD: rating NEG_PREFIX: neg_ LABEL_FIELD: label val_interval:   rating: "[0,3);(3,inf)" threshold:     rating: 4 load_col:     inter: [user_id, item_id, rating]     user: [user_id, age, gender, occupation]     item: [item_id, genre, release_year] # training and evaluation epochs: 500 train_batch_size: 4096 eval_batch_size: 40960000 eval_args:   split: {'RS':[0.8, 0.1, 0.1]}   group_by: ~   mode: labeled   order: RO valid_metric: AUC metrics: ['AUC', 'LogLoss']
 
您可以根据您的需求自行调整配置参数。

**Comment by NEUYuYang** [comment]
你好，我在用DIN模型跑ml-1m数据集
参数如下：
'field_separator': "\t",
&nbsp; &nbsp; 'seq_separator': " ",
&nbsp; &nbsp; 'data_path': "./dataset",
&nbsp; &nbsp; 'val_interval': {'rating': "[0,3);(3,inf)"},
&nbsp; &nbsp; 'threshold': {'rating': 4},
&nbsp; &nbsp; 'load_col': {'inter': ['user_id', 'item_id','timestamp','rating'],
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;'item': ['item_id', 'genre', 'release_year'],
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;'user': ['user_id', 'age', 'gender', 'occupation']}, 
&nbsp; &nbsp; 'USER_ID_FIELD': 'user_id', 
&nbsp; &nbsp; 'ITEM_ID_FIELD': 'item_id',
&nbsp; &nbsp; 'RATING_FIELD': 'rating',
&nbsp; &nbsp; 'NEG_PREFIX': 'neg_',
&nbsp; &nbsp; 'LABEL_FIELD': 'label',
&nbsp; &nbsp; 'eval_args':{'split':{'RS':[0.8,0.1,0.1]},
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;'group_by':'none',
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;'mode': 'labeled',
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;'order': 'TO'},
&nbsp; &nbsp; 'show_progress': False,
&nbsp; &nbsp; 'epochs': 200,
&nbsp; &nbsp; 'train_batch_size': 4096,
&nbsp; &nbsp; 'eval_batch_size': 40960000,
&nbsp; &nbsp; 'valid_metric': 'AUC',
&nbsp; &nbsp; 'metrics': ['AUC', 'LogLoss'],
&nbsp; &nbsp; 'embedding_size': 16,
&nbsp; &nbsp; 'learning_rate': 1e-2,
&nbsp; &nbsp; 'dropout_prob': 0.1,
&nbsp; &nbsp; 'mlp_hidden_size': '[128,128,128]',
&nbsp; &nbsp; 'pooling_mode': 'mean',
&nbsp; &nbsp; 'stopping_step': 200


目前的问题是，验证集的AUC最高值在第一个epoch并且只有0.62，其他epoch的验证AUC都在0.62以下，训练集损失是下降的
可否解惑一下哪里出错了


于洋
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "RUCAIBox/RecBole"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年11月16日(星期四) 下午5:44
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [RUCAIBox/RecBole] [🐛BUG] DIN AUC性能测试与原文不符 (Issue #1914)





 
@NEUYuYang
 您好！跑CTR任务需要指定数据的label。不进行特殊设置的话数据的所有交互都会被默认为正例。您可以通过设置threshold参数格局rating值区分正负例，具体可以参考我们的文档。一个例子是：
 field_separator: "\t" seq_separator: " " USER_ID_FIELD: user_id ITEM_ID_FIELD: item_id RATING_FIELD: rating NEG_PREFIX: neg_ LABEL_FIELD: label val_interval:   rating: "[0,3);(3,inf)" threshold:     rating: 4 load_col:     inter: [user_id, item_id, rating]     user: [user_id, age, gender, occupation]     item: [item_id, genre, release_year] # training and evaluation epochs: 500 train_batch_size: 4096 eval_batch_size: 40960000 eval_args:   split: {'RS':[0.8, 0.1, 0.1]}   group_by: ~   mode: labeled   order: RO valid_metric: AUC metrics: ['AUC', 'LogLoss']
 
您可以根据您的需求自行调整配置参数。
